Providers/Profile.swift

```diff
@@ -303,9 +306,33 @@ open class BrowserProfile: Profile {
         // By default, filter logging from Rust below `.info` level.
         try? RustLog.shared.setLevelFilter(filter: .info)
 
-        // This has to happen prior to the databases being opened, because opening them can trigger
-        // events to which the SyncManager listens.
-        self.syncManager = BrowserSyncManager(profile: self)
+        if prefs.intForKey(PrefsKeys.SecondProfileLaunchKey) != 1 {
+            // never been installed or launched v112 or a version with BrowserSyncManager
+            let msg = "Setting `syncManager` property to `BrowserSyncManager"
+            logger.log(msg, level: .debug, category: .sync)
+
+            self.syncManager = BrowserSyncManager(profile: self)
+
+            // update prefs so that when we try again next time we know its our 2nd launch
+            prefs.setInt(1, forKey: PrefsKeys.SecondProfileLaunchKey)
+        } else {
```

Comment:
This is very true that we don't need this considering on AppLaunch in AppDelegate we have proper values. 

Although I do have a question, if a user uses the `BrowserSyncManageron` 1st launch but gets `RustSyncManager` on Launch 2nd and for some reason they get `BrowserSyncManager` would that brick the app or cause issues?

I agree this is overkill but trying to avoid failure points for users here